### ISSUE TITLE
Chokes on C# 7 ref feature

### ISSUE THREAD

**Issue by darfink** [DarkCloud14/MonoDevelop.StyleCop#22]
C# version 7 allows reference to values types to be returned.
Code such as this leads to a parser error:
```c#
    ref Vector2 Position { get; }
```

> Warning: SA0102 : CSharp.CsParser : A syntax error has been discovered

**Comment by DarkCloud14** [comment]
As this comes from the original StyleCop and not the addin itself I'll check if it's fixed there already with a newer version. If it's fixed I'll update the used StyleCop version of this addin.

**Comment by DarkCloud14** [comment]
@darfink Sorry but even the latest available Beta of StyleCop on NuGet has the same problem.
I think the only way to get this fixed is to switch from Classic StyleCop to the new StyleCopAnalyzers project which won't be easy and might only be possible for the latest Releases of MonoDevelop or better Visual Studio for Mac as StyleCopAnalyzers uses Roslyn..

**Comment by darfink** [comment]
Alright, no worries. Let's just hope they it gets fixed in 5.0.0

**Comment by DarkCloud14** [comment]
Interesting, I just tested the latest StyleCop extension in VS2017 Community Edition (Windows) and there it works.. This is a bit strange as the error on VS for Mac is reported by StyleCop at least it looks like that.
I have to check that again, maybe I forgot to replace a file during the test of the Beta version in VS for Mac.  I keep you updated.

**Comment by DarkCloud14** [comment]
@darfink Ok I just checked the latest beta branch on GitHub and they added ref support for methods + locals but not for properties.

I've created an issue and also forked the current branch, fixed the problem for properties and send a pull request. If they accept it, it'll be fixed if not they hopefully fix it somehow else..

**Comment by DarkCloud14** [comment]
Ok my fix was merged so as soon as the next 5.0.0 beta or release is out it'll work.

**Comment by DarkCloud14** [comment]
Ok I plan to get this fixed with the next version which then also integrates a recent StyleCop version which fixes this problem.

**Comment by DarkCloud14** [comment]
The new version with a fix for this issue is finally done and can be downloaded at the releases page of the project at GitHub. Unfortunately the addin page of MonoDevelop doesn't work correctly anymore and so I can't update the version there so that it's automatically published to everybody who uses MonoDevelop addin update mechanism.

You must download the appropriate package and install it manually via install file in addin manager.
